### ISSUE TITLE
Serverless errorMessage: Cannot find module in production on Netlify 

### ISSUE THREAD

**Issue by solution-loisir** [11ty/eleventy#2138]
**Describe the bug**
Although it's working perfectly fine in development using `netlify-cli 8.1.7`, when I try to invoke the serverless function in production it fails and return this message:
```json
{"errorType":"Error","errorMessage":"Cannot find module '/var/task/functions/getcode/eleventy-serverless-map.json'\nRequire stack:\n- /var/task/functions/getcode/index.js\n- /var/task/getcode.js\n- /var/runtime/UserFunction.js\n- /var/runtime/index.js","trace":["Error: Cannot find module '/var/task/functions/getcode/eleventy-serverless-map.json'","Require stack:","- /var/task/functions/getcode/index.js","- /var/task/getcode.js","- /var/runtime/UserFunction.js","- /var/runtime/index.js","    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)","    at Function.Module._load (internal/modules/cjs/loader.js:746:27)","    at Module.require (internal/modules/cjs/loader.js:974:19)","    at require (internal/modules/cjs/helpers.js:93:18)","    at Serverless.getContentMap (/var/task/functions/getcode/index.js:96618:26)","    at Serverless.isServerlessUrl (/var/task/functions/getcode/index.js:96629:31)","    at Runtime.handler (/var/task/functions/getcode/index.js:116025:14)","    at processTicksAndRejections (internal/process/task_queues.js:95:5)"]}
```
I store the serverless function in functions/getcode/index.js.
**To Reproduce**
Might prove hard to reproduce.

**Environment:**

- Linux Mint 20
- @11ty/eleventy: 1.0.0-canary.46
- netlify-cli: 8.1.7
- Node v14
- NPM v6

**Additional context**
https://github.com/solution-loisir/web-x-ray

**Comment by AleksandrHovhannisyan** [comment]
If I understood the solution as described in the thread here (linked in a response to your Netlify forum post): https://githubmemory.com/repo/11ty/eleventy/issues/1866

It looks like this was just an issue of not using `exports.handler`, right?

**Comment by solution-loisir** [comment]
> It looks like this was just an issue of not using exports.handler, right?

Hi @AleksandrHovhannisyan, yes in that case it was. Although, I didn't change that part of the auto-generated function so I do use `exports.handler`. I'll keep testing and post back if I find a solution. Thank you for commenting. I probably made a mistake somewhere, but it might also be a bug.

**Comment by solution-loisir** [comment]
I finally fixed the error. I removed the `node_bundler = esbuild` option from the [`[functions]`](https://docs.netlify.com/configure-builds/file-based-configuration/#functions) section in `netlify.toml` config file and that solved it! Maybe the `esbuild` bundler method used in [@netlify/zip-it-and-ship-it](https://github.com/netlify/zip-it-and-ship-it) library is somehow conflicting with Eleventy (would need to be confirmed). Although, it's very probable that I just misused this configuration. Any how, I'm glad I could fix my little learning project (https://webxray.netlify.app/)!
